container/lxd/lxd.go

```diff
@@ -140,6 +157,8 @@ func (manager *containerManager) CreateContainer(
 		"boot.autostart": "true",
 	}
 
+	ApplyConstraints(metadata, cons)
```

Comment:
I sent an email about this to Juju-dev.  The problem is that our constraints and lxd's limits are different concepts, and I'm not sure if it's really possible to reconcile the two in any kind of way that's not horrible.